force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -54,50 +60,78 @@ public with sharing class FlowControlLoopsCollections {
      * @param num2 The second number.
      * @return The maximum of the two numbers, or null if either number is null.
      */
-    public static Integer findMax(Integer num1, Integer num2) {
-        // Initialize largestNum as null
-        Integer largestNum = null;
-
-        return null; // Replace null with the variable you used to store the result
-    }
-
+        public static Integer findMax(Integer num1, Integer num2) {
+            //Check if either num1 or num2 is null
+            if (num1 == null || num2 == null){
+                return null;
+
+            }
+            Integer largestNum;
+            if (num1 > num2){
+                largestNum = num1;
+            } else {
+                largestNum = num2;
+            }
+            return largestNum;
+        }
 
-    /**
+    /** Q4
      * Question 4
      * Checks if a number is positive, negative, or zero.
      * If the number is null, return null.
      * Example: checkNumber(5) should return "Positive"
      * @param a The number to check.
      * @return A string indicating whether the number is "Positive", "Negative", or "Zero", or null if the number is null.
      */
-    public static String checkNumber(Integer a) {
-        return null; // Replace null with the variable you used to store the result
-    }
+        public static String checkNumber(Integer a) {
+        if (a == null) {
+            return null;
+        }
+        if (a > 0) {
+            return 'Positive';
+        } else if (a < 0) {
+            return 'Negative';
+        } else {
+            return 'Zero';
+        }
+        }
 
-    /**
+    /** Q5
      * Question 5
      * Checks if a number is even or odd.
      * If the number is null, return null.
      * Example: checkEvenOdd(2) should return "Even"
      * @param a The number to check.
      * @return A string indicating whether the number is "Even" or "Odd", or null if the number is null.
      */
-    public static String checkEvenOdd(Integer a) {
-        return null; // Replace null with the variable you used to store the result
-    }
+        public static String checkEvenOdd(Integer a) {
+        if (a == null) {
+            return null;
+        } else if (Math.mod(a, 2) == 0) {
+            return 'Even';
+        } else {
+            return 'Odd';
+        }
+        }
 
-    /**
+    /** Q6
      * Question 6
      * Checks if a string is empty, null, or contains text.
      * Example: checkString("") should return "Empty"
      * @param a The string to check.
      * @return A string indicating whether the input is "Empty", "Null", or "Contains Text".
      */
-    public static String checkString(String a) {
-        return null; // Replace null with the variable you used to store the result
-    }
+        public static String checkString(String a) {
+        if (a == null) {
+            return 'Null';
+        } else if (a.length() == 0) {
+            return 'Empty';
```

Comment:
This is a clever way to approach the problem. But there's also a String method called **[isEmpty()](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_string.htm#apex_System_String_isEmpty)** that would be helpful here.